src/replica/test/open_replica_test.cpp

```diff
@@ -0,0 +1,118 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <gtest/gtest.h>
+
+#include "replica/replica.h"
+#include "mock_utils.h"
+#include "meta/meta_data.h"
+#include "replica_test_base.h"
+
+namespace dsn {
+namespace replication {
+
+class open_replica_test : public replica_test_base
+{
+public:
+    open_replica_test() = default;
+    ~open_replica_test() { dsn::utils::filesystem::remove_path("./tmp_dir"); }
+
+    std::unique_ptr<mock_replica> create_fake_replica(gpid gpid, app_info &app_info, ballot ballot, decree decree, partition_status::type status, int partition_num)
```

Comment:
Base class `replica_test_base` will add an replica defaultly in its ctor, if it's not what you need, do not add it.
https://github.com/XiaoMi/rdsn/blob/4ee361b3fc5de7073b95fce265c3870b926ac841/src/replica/test/replica_test_base.h#L56